src/servers/src/http/prometheus.rs

```diff
@@ -199,7 +257,64 @@ pub async fn instant_query(
         .with_label_values(&[query_ctx.get_db_string().as_str(), "instant_query"])
         .start_timer();
 
-    let result = handler.do_query(&prom_query, query_ctx).await;
+    if let Some(name_matchers) = find_metric_name_not_equal_matchers(&promql_expr)
```

Comment:
Any chance to put this into `do_query` for better test-ability? We better keep this http layer as thin as possible.

Reply:
I've attempted this before, but it complicates the code. I believe it's acceptable to have this logical query in a loop that collects responses; the actual logic resides within the `do_query`.